### DISCUSSION TITLE
scikit classifier not fitting on model

### DISCUSSION THREAD

**Discussion by james6601** [Trusted-AI/adversarial-robustness-toolbox#2115] [Q&A]
Hi everyone, I am trying to fit my model with the `art.estimators.sklearnclassifer` but it only recognises one class label even though i have 2 [0,1]. Can anyone at all help me resolve this issue?

full error with traceback:
```
Traceback (most recent call last):


  File ~\AppData\Roaming\Python\Python39\site-packages\art\estimators\classification\classifier.py:73 in replacement_function
    return fdict[func_name](self, *args, **kwargs)

  File ~\AppData\Roaming\Python\Python39\site-packages\art\estimators\classification\scikitlearn.py:176 in fit
    self.model.fit(x_preprocessed, y_preprocessed, **kwargs)

  File C:\ProgramData\Anaconda3\lib\site-packages\sklearn\svm\_base.py:199 in fit
    y = self._validate_targets(y)

  File C:\ProgramData\Anaconda3\lib\site-packages\sklearn\svm\_base.py:720 in _validate_targets
    raise ValueError(

ValueError: The number of classes has to be greater than one; got 1 class
```

EDIT 

I have encoded my train labels as well as my test labels and I'm still receiving the error
```
np.unique(y_train)
Out[3]: array([0, 1])
```

@beat-buesser can you advise?

**Comment by beat-buesser** [comment]
Hi @james6601 Is `y_trn` and `y_train` the same?

**Reply by james6601** [reply]
Hi @beat-buesser sorry I changed `y_trn` to `y_train` but I'm still receiving the error. Its as if the ART classifier doesnt recognise the different class labels

**Reply by beat-buesser** [reply]
Are you providing the labels as strings BENIGN and DDoS? If yes, you have to convert it into a one-hot-encoded array.

**Reply by james6601** [reply]
Hi, @beat-buesser.
Yeah i have one hot encoded the labels aswell as encoding them before hand but now receiving an error saying that axis 1 is out of bounds for dimension 1

**Reply by beat-buesser** [reply]
Hi @james6601 Can you show more details?
